### PR TITLE
Gh477 fallback policy loop

### PR DESCRIPTION
**Pull request by naoko**
**Proposed changes**:
- Fallback policy could trip cirucuit breaker if subsequent intent is
predicted lower than threshold thus added the condition to prevent
predict twice in a row.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by imsurinder90** [comment]
FYI, There is an easiest way to fix the issue where fallback policy is repeated number of times. The below code will run only if the `latest_action_name == 'action_listen'` 

```
        score = self.core_threshold
        from rasa_core.actions.action import ACTION_LISTEN_NAME
        if tracker.latest_action_name == ACTION_LISTEN_NAME:
            if nlu_confidence < self.nlu_threshold:
                logger.debug("NLU confidence {} is lower "
                             "than NLU threshold {}. "
                             "Predicting fallback action: {}"
                             "".format(nlu_confidence, self.nlu_threshold,
                                       self.fallback_action_name))
                score = 1.0
        result[idx] = score

        return result
```
Thanks

**Comment by naoko** [comment]
@imsurinder90 , Thank you, I will update accordingly

File: rasa_core/policies/fallback.py [hunk 1]
```diff
@@ -63,6 +63,17 @@ def train(self,
...
+    def should_fallback(self,
+                        nlu_confidence,  # type float
+                        last_action_name  # type: Text
+                        ):
+        # type: (...) -> bool
+        """It should predict fallback action only if
+        a. predicted NLU confidence is lower than ``nlu_threshold`` &&
+        b. last action is NOT fallback action
+        """
+        return nlu_confidence < self.nlu_threshold and last_action_name != self.fallback_action_name  # noqa
```
Comment (tmbo) [inline rasa_core/policies/fallback.py]: I don't think we should put a `noqa` here.

Comment (naoko) [reply rasa_core/policies/fallback.py]: Got it. Removed #qa and broke up into 2 lines to comply PEP8

**Comment by paulhopkins** [comment]
I have also had issues with the fallback repeating, so I tried the two suggestions here. However, I found it would cycle between `utter_default` and another action, or would jump about to other parts of the conversation. Can we force it to perform the action and then listen using something like `tracker.trigger_follow_up_action(ActionListen())`. Would this need a custom action?

**Pull request merged**